dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/tei/DefaultTeiAnalyticsDimensionsService.java

```diff
@@ -73,37 +66,29 @@ class DefaultTeiAnalyticsDimensionsService implements TeiAnalyticsDimensionsServ
     private final ProgramService programService;
 
     @Override
-    public List<PrefixedDimension> getQueryDimensionsByTrackedEntityTypeId( String trackedEntityTypeId )
+    public List<PrefixedDimension> getQueryDimensionsByTrackedEntityTypeId( String trackedEntityTypeId,
+        Collection<String> programUids )
```

Comment:
Instead of using Collection, I would use something more specific, like Set in this case (as the program cannot be repeated).

Reply:
done